helix-core/src/test/java/org/apache/helix/controller/stages/TestPerReplicaThrottleStage.java

```diff
@@ -100,6 +115,246 @@ public void testNoThrottleMessagePass() {
     Assert.assertTrue(msg.getId().equals("001"));
   }
 
+  // case 0. N1(O), N2(S), N3(O), message N3(O->S) is treated as recovery
```

Comment:
I was thinking a test case that can validate the following scenario:
Current states: N1(S), N2(O), N3(O)
BestPossible: N1(M), N2(S), N3(S)
Only one message is considered as recovery rebalance based on your defined priority rule. 
Do you think this can be testified in this PR? You current test cases do not seem to be testing this.

Reply:
added as `testOneRecoveryOneLoad`.